pkg/pillar/cmd/zedrouter/probe.go

```diff
@@ -183,28 +191,28 @@ func niProbingUpdatePort(ctx *zedrouterContext, port types.NetworkPortStatus,
 // after port or NI changes, if we don't have a current uplink,
 // randomly assign one and publish, if we already do, leave it as is
 // each NI may have a different good uplink
-func checkNIprobeUplink(ctx *zedrouterContext, status *types.NetworkInstanceStatus) {
+func checkNIprobeUplink(ctx *zedrouterContext, status *types.NetworkInstanceStatus, intf string) {
 	// find and remove the stale info since the port has been removed
 	for _, info := range status.PInfo {
+		if info.IfName != intf {
+			continue
+		}
+		log.Infof("checkNIprobeUplink: %s, intf %s, is present %v\n", status.BridgeName, info.IfName, info.IsPresent)
 		if !info.IsPresent {
 			if _, ok := status.PInfo[info.IfName]; ok {
+				log.Infof("checkNIprobeUplink: %s remove intf %s from ProbeInfo\n", status.BridgeName, info.IfName)
 				delete(status.PInfo, info.IfName)
+				publishNetworkInstanceStatus(ctx, status)
 			}
-		} else {
-			info.IsPresent = false
 		}
 	}
 
 	if status.CurrentUplinkIntf != "" {
-		if _, ok := status.PInfo[status.CurrentUplinkIntf]; ok {
-			if strings.Compare(status.PInfo[status.CurrentUplinkIntf].IfName, status.CurrentUplinkIntf) == 0 {
-				return
-			}
-		}
-		// if the Current Uplink intf does not have an info entry, re-pick one below
-		status.CurrentUplinkIntf = ""
+		// regardless of the Curr Uplink is valid or not, it was the previous run result, let probing code to handle
+		return
```

Comment:
With the example above, assuming network instance is using eth2 to begin with. And the eth2 is moved out of device network status. NI will continue using eth2 till probing decides that eth2 is dead for us, right?

Reply:
No, it will not. Since the Pinfo of the NI does not have the 'eth2' map entry anymore. So the probing is going to pick a different one now. But we want to keep the previous "CurrentUPlinkIntf" unchanged, so by the end of the probing, we know, the previous one was 'eth2', and now is either 'eth0' or 'eth1'. If we change the "CurrentUPlinkIntf" inside the checkNIprobeUplink() as the code before, then we lost the previous intf information.